python/cudf/cudf/core/column_accessor.py

```diff
@@ -604,6 +604,17 @@ def _select_by_label_slice(self, key: slice) -> Self:
         if key.step is not None:
             raise TypeError("Label slicing with step is not supported")
 
+        if len(self.names) == 0:
+            # OK to slice empty columns
+            # https://github.com/rapidsai/cudf/issues/18376
+            return type(self)(
+                {},
+                multiindex=self.multiindex,
```

Comment:
Could you include `rangeindex=self.rangeindex`?

Or `return self._from_columns_like_self([], verify=False)` would do something similar